### PR TITLE
refactor(app): replace RobotWorkSpace in DeckThumbnail with BaseDeck

### PR DESCRIPTION
**Pull request by koji**
# Overview
replace RobotWorkSpace with BaseDeck
- ChooseProtocolSlideout
- ProtocolCard
- ProtocolDetails
- ProtocolDetails for ODD

For LabwareInfoOverlay, the component requires `runId` since `LabwareInfo` uses `runId` for `useLabwareOffsetForLabware`.
`useLabwareOffsetForLabware` uses `runId` to get `mostRecentAnalysis`.

close RAUT-819

# Test Plan
open the above components on Desktop app and Touchscreen app

# Changelog
- update DeckThumbnail component and its test
- add a prop(protocolAnalysis) to the above 4 components 

# Review requests

# Risk assessment
low

### CODE REVIEW

File: app/src/molecules/DeckThumbnail/index.tsx [hunk 1]
```diff
@@ -1,179 +1,125 @@
...
   )
-  const initialLoadedModulesBySlot = parseInitialLoadedModulesBySlot(commands)
-  const initialLoadedLabwareByModuleId = parseInitialLoadedLabwareByModuleId(
-    commands
+  const attachedModules =
+    useAttachedModules({
+      refetchInterval: ATTACHED_MODULE_POLL_MS,
+    }) ?? []
+
+  const enableDeckConfig = useFeatureFlag('enableDeckConfiguration')
```
Comment (brenthagen) [inline app/src/molecules/DeckThumbnail/index.tsx]: no need to use the deck config feature flag, will be removed soon

File: app/src/molecules/DeckThumbnail/index.tsx [hunk 2]
```diff
@@ -1,179 +1,116 @@
...
-  DeckSlot,
+  CompletedProtocolAnalysis,
   Liquid,
   LoadedLabware,
+  ProtocolAnalysisOutput,
   RunTimeCommand,
 } from '@opentrons/shared-data'
 
 interface DeckThumbnailProps extends StyleProps {
+  protocolAnalysis?: CompletedProtocolAnalysis | ProtocolAnalysisOutput | null
```
Comment (brenthagen) [inline app/src/molecules/DeckThumbnail/index.tsx]: should make `protocolAnalysis` required

File: app/src/molecules/DeckThumbnail/index.tsx [hunk 3]
```diff
@@ -1,179 +1,116 @@
...
   commands: RunTimeCommand[]
   labware: LoadedLabware[]
   liquids?: Liquid[]
```
Comment (brenthagen) [inline app/src/molecules/DeckThumbnail/index.tsx]: think we don't need these props, we can get them from `protocolAnalysis`

**Review by brenthagen — COMMENTED** [review]
looks good so far, but we need to add liquid info also. we don't have an example `BaseDeck` implementation of liquids yet - will be similar to RAUT-821

**Comment by koji** [comment]
- [ ] add liquids

File: app/src/molecules/DeckThumbnail/index.tsx [hunk 4]
```diff
@@ -1,179 +1,258 @@
...
-  showSlotLabels?: boolean
+  protocolAnalysis: CompletedProtocolAnalysis | ProtocolAnalysisOutput | null
 }
 
-export function DeckThumbnail(props: DeckThumbnailProps): JSX.Element {
-  const {
-    commands,
-    liquids,
-    labware = [],
-    showSlotLabels = false,
```
Comment (brenthagen) [inline app/src/molecules/DeckThumbnail/index.tsx:52]: we want leave this prop in, and defaulted to false to hide the `SlotLabels`. only the ODD protocol details map version of `DeckThumbnail` shows slot labels.

File: app/src/molecules/DeckThumbnail/index.tsx [hunk 5]
```diff
@@ -1,179 +1,258 @@
...
+    const topLabwareDefinition =
+      labwareInAdapterInMod?.result?.definition ?? module.nestedLabwareDef
+    return {
+      moduleModel: module.moduleDef.model,
+      moduleLocation: { slotName: module.slotName },
+      innerProps:
+        module.moduleDef.model === THERMOCYCLER_MODULE_V1
+          ? { lidMotorState: 'open' }
+          : {},
+      nestedLabwareDef: topLabwareDefinition,
```
Comment (brenthagen) [inline app/src/molecules/DeckThumbnail/index.tsx:90]: here we want to add the well fill to the nested labware - add a prop `nestedLabwareWellFill` to `moduleLocations` in `BaseDeck`

File: app/src/molecules/DeckThumbnail/index.tsx [hunk 6]
```diff
@@ -1,179 +1,258 @@
...
+        labwareInAdapter?.result?.definition ?? labwareDef
+      const topLabwareId = labwareInAdapter?.result?.labwareId ?? labwareId
+      const topLabwareDisplayName =
+        labwareInAdapter?.params.displayName ?? displayName
+
+      return {
+        labwareLocation: { slotName },
+        definition: topLabwareDefinition,
+        topLabwareId,
+        topLabwareDisplayName,
```
Comment (brenthagen) [inline app/src/molecules/DeckThumbnail/index.tsx]: simliar, we want to add the well fill to the labware - add a prop `wellFill` to `labwareLocations` in `BaseDeck`

File: app/src/molecules/DeckThumbnail/index.tsx [hunk 7]
```diff
@@ -1,179 +1,258 @@
...
-            const labwareInSlot =
-              slotId in initialLoadedLabwareBySlot
-                ? initialLoadedLabwareBySlot[slotId]
-                : null
-            const labwareInModule =
-              moduleInSlot?.result?.moduleId != null &&
-              moduleInSlot.result.moduleId in initialLoadedLabwareByModuleId
-                ? initialLoadedLabwareByModuleId[moduleInSlot.result.moduleId]
-                : null
+      {map(
```
Comment (brenthagen) [inline app/src/molecules/DeckThumbnail/index.tsx]: all of the children content here inside `BaseDeck` should not not be needed if we are processing `moduleLocations` and `labwareLocations` correctly above - we should be able to use

```
          const wellFill = getWellFillFromLabwareId(
            topLabwareId ?? '',
            liquids,
            labwareByLiquidId
          )
          const labwareHasLiquid = !isEmpty(wellFill)
```
in those two locations above with the added well fill props

**Review by brenthagen — COMMENTED** [review]
with these changes, we should end up with a fairly simple `<BaseDeck {...someProps} />` instance (with no children) in the render of `DeckThumbnail`

**Comment by brenthagen** [comment]
also, it would be a good idea to change the title of this PR to indicate it is only doing a replacement in DeckThumbnail

File: app/src/molecules/DeckThumbnail/index.tsx (hunk 7 above)
Comment (koji) [reply app/src/molecules/DeckThumbnail/index.tsx]: @brenthagen 
updated code based on your comment and one question about `labwareHasLiquid`.
Do we need to add interactions that only exists in DeckView of `SetupLiquidsMap`? 

Screenshot 2023-11-06 at 1 47 25 PM

Comment (brenthagen) [reply app/src/molecules/DeckThumbnail/index.tsx]: i think no, we don't need to add liquid hover or click interactions to DeckThumbnail. so we won't need `labwareHasLiquid` in DeckThumbnail

Comment (koji) [reply app/src/molecules/DeckThumbnail/index.tsx]: okay, thank you!

File: app/src/molecules/DeckThumbnail/index.tsx [hunk 8]
```diff
@@ -1,179 +1,129 @@
...
-                  </g>
-                ) : null}
-              </React.Fragment>
-            )
-          })}
-          {showSlotLabels ? <SlotLabels robotType={robotType} /> : null}
-        </>
-      )}
-    </RobotWorkSpace>
+      {showSlotLabels ? <SlotLabels robotType={robotType} /> : null}
```
Comment (brenthagen) [inline app/src/molecules/DeckThumbnail/index.tsx]: `SlotLabels` can actually be removed in this component, `showSlotLabels` added as a prop to `BaseDeck` and passed there

File: components/src/hardware-sim/BaseDeck/BaseDeck.tsx [hunk 9]
```diff
@@ -184,21 +191,30 @@ export function BaseDeck(props: BaseDeckProps): JSX.Element {
...
               labwareLocation !== 'offDeck' &&
               'slotName' in labwareLocation &&
               s.id === labwareLocation.slotName
           )
+          const labwareHasLiquid = !isEmpty(wellFill)
           return slotDef != null ? (
             <g
               key={slotDef.id}
               transform={`translate(${slotDef.position[0]},${slotDef.position[1]})`}
+              cursor={labwareHasLiquid ? 'pointer' : ''}
```
Comment (brenthagen) [inline components/src/hardware-sim/BaseDeck/BaseDeck.tsx]: would be better to condition cursor on existence of `onLabwareClick`

**Review by brenthagen — APPROVED** [review]
looks good, comments on slot labels and hover could be addressed in a follow on PR if wanted

**Pull request merged**